clang/lib/Headers/immintrin.h

```diff
@@ -580,8 +580,7 @@ _storebe_i64(void * __P, long long __D) {
 #include <cetintrin.h>
 #endif
 
-/* Some intrinsics inside adxintrin.h are available only on processors with ADX,
- * whereas others are also available at all times. */
```

Comment:
Sorry, just noticed this. Since we removed the comments, should we use the same format for `adxintrin.h` and add comment for `adcintrin.h`, e.g., 

```
/* Intrinsics inside adcintrin.h are available at all times. */
#include <adcintrin.h>

#if !(defined(_MSC_VER) || defined(__SCE__)) || __has_feature(modules) ||      \
    defined(__ADX__)
#include <adxintrin.h>
#endif
```

Reply:
> add comment for adcintrin.h

Up to you. I felt the header names imply that `<adxintrin.h>` has adx specific intrinsics and `<adcintrin.h>` has the always available adc intrinsics.

Let me know and I can add a, "/* Intrinsics inside adcintrin.h are available at all times. */", comment above `<adcintrin.h>`.

> should we use the same format for adxintrin.h

I didn't protect `<adxintrin.h>` since it was unprotected earlier and didn't want to introduce a behavioural change there.

For `_MSC_VER` this header isn't large enough to cause the current compile-time issues with MSVC STL.

For `__SCE__` the git blame here, https://github.com/llvm/llvm-project/commit/046130490f230559e8dae7d7cb4959bd04f89bb8, appears to have missed guarding this header to prevent accidental usage of these intrinsics on PS4 since Intel ADX isn't available on the PS4 Jaguar chips.

Reply:
I think comment is good. Without it, people may wonder anything special here.
For ADX, I guess the reason is it mixed with adc intrinsics previously, so cannot be guarded. And we should be free to do it now.

Reply:
Confirmed that Intel ADX is supported on Ryzen.

I checked the Sony PS5 compiler on my work pc and confirmed that `__ADX__` is defined to `1` as expected for the PS5 Ryzen chips so we won't break them.
I check the Sony PS4 compiler on my work pc and confirmed that `__ADX__` is not defined as expected for the PS4 Jaguar chips.